daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Pretty.scala

```diff
@@ -181,7 +181,7 @@ private[lf] object Pretty {
             )
           case DisclosurePreprocessing.NonExistentTemplate(templateId) =>
             text(
-              s"Template $templateId does not exist"
+              s"Template $templateId does not exist for the disclosed contract"
             )
```

Comment:
Something I missed in a previous PR.

I would better replace 
```
final case class NonExistentTemplate(templateId: TypeConName) extends Error
```
by 
```
   final case class LookupError(error: language.LookupError) extends Error
```
and replace the lines here by something like:
```
          case LookupError(error) =>
            text(s"${error.pretty} for disclosed contract")
```

Reasons being:
1 - error.pretty has better error messages
2 - we get more consistent error messages about loopuk error message
3 - we will not need to extend  `DisclosurePreprocessing.Error` in the future if the disclosure preprocessing need more info.

Reply:
side question, do we check contracts provided with key hash are actual of a type with contract key.

Reply:
> side question, do we check contracts provided with key hash are actual of a type with contract key.

https://github.com/digital-asset/daml/issues/13897 exists to generally investigate validation of disclosed key hashes.

This PR introduces changes that do check that the contract key type, the operation template type and the contract type all match (ref: https://github.com/digital-asset/daml/blob/145a5da6cc723694a5b210a0ba7782f0d552ee11/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala#L1547)

Or did you have some other type of validation in mind that I'm missing here?